server/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -610,6 +610,85 @@ public void testToXContent() throws IOException {
         }
     }
 
+    public void testGenerateFailureToXContentWithNoDetails() throws IOException {
+        {
+            Exception ex;
+            if (randomBoolean()) {
```

Comment:
Perhaps a better test would be to do random level of wrapping (up to 10), e.g. a loop of n times that wraps one exception to another.

Reply:
Good idea